### PR TITLE
[Move Expando]Lps 190836 11 23 3

### PR DESCRIPTION
**Pull request by slnn**
https://liferay.atlassian.net/browse/LPS-190836

### CODE REVIEW

**Comment by slnn** [comment]
Compare with this https://github.com/liferay-core-infra/liferay-portal/pull/6141, I just updated the upgrade part to make it more consistent with other modules

I tested the upgrade process with the 62 db dump locally, it worked well.

1. Start portal with an empty db named fresh;
2. Stop portal 
3. Export the db dump by running ant -f build-test.xml export-database-schema -Ddatabase.file.prefix=fresh -Ddatabase.schema.name=fresh
4. Create another new empty db named 62x and import the 62 db dump:[data-archive-portal-mysql.zip](https://github.com/liferay-core-infra/liferay-portal/files/13447793/data-archive-portal-mysql.zip)
5. Run upgrade process with the db 62x
6. Start portal with the upgraded db 62x
7. Stop portal
8.  Export the db dump by running ant -f build-test.xml export-database-schema -Ddatabase.file.prefix=upgraded -Ddatabase.schema.name=62x
9. Running ant -f build-test.xml validate-db-upgrade-schemas -Dfresh.db.schema=fresh -Dupgraded.db.schema=62x after making sure ${liferay.home}/poshi/upgraded-62x.sql and ${liferay.home}/poshi/fresh-fresh.sql are not empty and contain real data

Here are the details:

The fresh db dump:
[fresh-fresh.sql.zip](https://github.com/liferay-core-infra/liferay-portal/files/13447757/fresh-fresh.sql.zip)

The upgraded db dump:
[upgraded-62x.sql.zip](https://github.com/liferay-core-infra/liferay-portal/files/13447760/upgraded-62x.sql.zip)

Result of running "ant -f build-test.xml validate-db-upgrade-schemas -Dfresh.db.schema=fresh -Dupgraded.db.schema=62x"

[62x-fresh-validate-db-schemas.txt](https://github.com/liferay-core-infra/liferay-portal/files/13447762/62x-fresh-validate-db-schemas.txt)

Upgrade log:
[62-to-pr-upgrade.txt.zip](https://github.com/liferay-core-infra/liferay-portal/files/13447864/62-to-pr-upgrade.txt.zip)

Portal log with upgraded db:
[62-to-pr-catalina.out.zip](https://github.com/liferay-core-infra/liferay-portal/files/13447868/62-to-pr-catalina.out.zip)

**Comment by slnn** [comment]
Hi @yuhai 

Could you review the upgrade part, please?

Thanks
Lily

**Comment by slnn** [comment]
Forced push to resolve conflicts

**Comment by slnn** [comment]
ci:test:sf

File: modules/apps/expando/expando-service/service.xml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0"?>
 <!DOCTYPE service-builder PUBLIC "-//Liferay//DTD Service Builder 7.4.0//EN" "http://www.liferay.com/dtd/liferay-service-builder_7_4_0.dtd">
 
-<service-builder auto-import-default-references="false" change-tracking-enabled="true" mvcc-enabled="true" package-path="com.liferay.expando">
+<service-builder auto-import-default-references="false" change-tracking-enabled="true" dependency-injector="ds" mvcc-enabled="true" package-path="com.liferay.expando">
 	<namespace>Expando</namespace>
 	<entity local-service="true" name="ExpandoColumn" remote-service="true">
```
Comment (yuhai) [inline modules/apps/expando/expando-service/service.xml:6]: @slnn, based on the doc from liferay-service-builder_7_4_0.dtd, the module is new. 

> For new OSGi modules, the default value is "ds"

It seems the commit is not necessary. Please verify this again. If it is necessary, I suggest we should provide the related explanation in the commit message.

File: modules/apps/portal-cache/portal-cache-ehcache-impl/src/main/resources/ehcache/liferay-multi-vm.xml [hunk 2]
```diff
@@ -219,23 +219,23 @@
...
 		overflowToDisk="false"
 		timeToIdleSeconds="600"
 	>
 	</cache>
 	<cache
 		eternal="false"
 		maxElementsInMemory="100000"
-		name="com.liferay.portal.kernel.dao.orm.FinderCache.com.liferay.portlet.expando.model.impl.ExpandoValueImpl.List2"
+		name="com.liferay.portal.kernel.dao.orm.FinderCache.com.liferay.expando.model.impl.ExpandoValueImpl.List2"
 		overflowToDisk="false"
```
Comment (yuhai) [inline modules/apps/portal-cache/portal-cache-ehcache-impl/src/main/resources/ehcache/liferay-multi-vm.xml:239]: @slnn, the file liferay-multi-vm.xml seems defined cache for core. After moving Expando to module, please add them to module-multi-vm.xml in the service module. Please see https://help.liferay.com/hc/en-us/articles/360035581451-Introduction-to-Cache-Configuration

**Comment by yuhai** [comment]
@slnn, please see the above comments. For the upgrade part, the fix placed `UpgradePackages ` to module. It might be late. For example, when upgrading the `com.liferay.portal.upgrade.v7_4_x.PortalUpgradeProcessRegistryImpl`. If `entiry.setExpandoBridgeAttributes(serviceContext)` executed, for example, `user.setExpandoBridgeAttributes(serviceContext)`. It will invoke `ExpandoColumnPermissionUtil.contains`, it will use `ExpandoColumn.class.getName()` to fetch permission data. At that time, the related packages didn't been updated. I think we should place `UpgradePackages ` to the core.

File: modules/apps/portal-cache/portal-cache-ehcache-impl/src/main/resources/ehcache/liferay-multi-vm.xml (hunk 2 above)
Comment (slnn) [reply modules/apps/portal-cache/portal-cache-ehcache-impl/src/main/resources/ehcache/liferay-multi-vm.xml:239]: Hi @yuhai 

From  liferay-multi-vm.xml, I can see this comment `"Caches in this file can be clustered and are configured to be clustered."`, so it is safe to move to module-multi-vm.xml in the service module?


Lily

Comment (slnn) [reply modules/apps/portal-cache/portal-cache-ehcache-impl/src/main/resources/ehcache/liferay-multi-vm.xml:239]: As we talked, I will move them into expano-service module

**Comment by slnn** [comment]
Hi @yuhai 

When digging into this potential risk find that it is impossible to call ExpandoColumnPermissionUtil.contains through entiry.setExpandoBridgeAttributes(serviceContext) in the core upgrade process which mean these upgrade process in https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portal/upgrade/v7_4_x/PortalUpgradeProcessRegistryImpl.java, because the expando service module is not ready during this period, however,  this is not caused by this pull as blog service module also did not ready on the master and this pull during this period, from gogo shell I can see so many unsatisfied references:
Expando service module: https://gist.github.com/slnn/c6ba114a88bc4f596ecd986f84f876fd
Blog service module:  https://gist.github.com/slnn/41f1de6b673bedfef55a03a90215766d

I resend two pull requests one added the upgrade process in core: https://github.com/liferay-core-infra/liferay-portal/pull/6321 another one added the new upgrade process in module:https://github.com/liferay-core-infra/liferay-portal/pull/6322, but I did not move the [UpgradeExpandoColumn](https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portal/upgrade/v7_4_x/PortalUpgradeProcessRegistryImpl.java#L132) which added in the core upgrade process by https://liferay.atlassian.net/browse/LRAC-10726  from core into the expando service module, do we need to move it?

> @slnn, please see the above comments. For the upgrade part, the fix placed `UpgradePackages ` to module. It might be late. For example, when upgrading the `com.liferay.portal.upgrade.v7_4_x.PortalUpgradeProcessRegistryImpl`. If `entiry.setExpandoBridgeAttributes(serviceContext)` executed, for example, `user.setExpandoBridgeAttributes(serviceContext)`. It will invoke `ExpandoColumnPermissionUtil.contains`, it will use `ExpandoColumn.class.getName()` to fetch permission data. At that time, the related packages didn't been updated. I think we should place `UpgradePackages ` to the core.

**Pull request closed without merging**